data_transfer/lib/dreem.py

```diff
@@ -58,39 +50,44 @@ def get_restricted_list(session: requests.Session, user_id: str) -> List[dict]:
     results = []
 
     while url:
-        response = session.get(url)
-        # TODO: catch/log exception
-        response.raise_for_status()
-        result: dict = response.json()
-        url = result["next"]
-        results.extend(result["results"])
+        try:
+            response = session.get(url)
+            response.raise_for_status()
+            result: dict = response.json()
+            url = result["next"]
+            results.extend(result["results"])
+        except requests.HTTPError:
+            # attempt to skip this one record
```

Comment:
Did you get this error before? Just curious - but will this ever jump out of the while loop if it keeps throwing? (As it potentially reuses the same `result["next"]`) Perhaps narrow the error down to the specific HTTPError?

Reply:
I received one of the derived errors (`RequestException` I think). `HTTPError` is similar to `Exception` in that it contains all requests specific potential errors. Good point -- the `result[next]` will be different in each iteration so won't be infinite-lopped, so in that case it'll catch as many errors as the length of results.